File: nutrinova-api/NutrinovaData/Features/Reports/PatientNutrientReport.cs
namespace NutrinovaData.Features.Reports;

public class PatientNutrientReport
{
  public string PatientId { get; set; } = string.Empty;

  public string PatientName { get; set; } = string.Empty;

  public int DaysAchievedCount { get; set; }

  public decimal MaxConsumption { get; set; }

  public decimal AvgConsumption { get; set; }

  public IEnumerable<DailyNutrientGoalReport> Days { get; set; } = default!;
}

File: nutrinova-ui/src/app/(authorized)/goals/_models/PatientNutrientReport.ts
import { DailyNutrientGoalReport } from "./NutrientGoalReportItem";


export interface PatientNutrientReport {
  patientId: string;
  patientName: string;
  daysAchievedCount: number;
  maxConsumption: number;
  avgConsumption: number;
  days: DailyNutrientGoalReport[];
}
